classes/listener/observer.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+namespace block_vmchecker\listener;
+
+defined('MOODLE_INTERNAL') || die();
+
+class observer {
+    private static function stopPreviousAttemp($assignmentId) {
+        global $DB, $USER, $CFG;
+        
+        $previousAttempt = $DB->get_record('block_vmchecker_submissions',
+            array(
+                'userid' => $USER->id,
+                'assignid' => $assignmentId,
+        ));
+        if (!$previousAttempt)
+            return;
+
+        $ch = curl_init();
+        curl_setopt($ch, CURLOPT_URL, $CFG->block_vmchecker_backend . $previousAttempt->uuid . '/cancel');
+        curl_setopt($ch, CURLOPT_HTTPHEADER, array("Content-Type: application/json"));
+        curl_setopt($ch, CURLOPT_POST, true);
+        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+
+        curl_exec($ch);
+
+        curl_close($ch);
+
+        $DB->delete_records('block_vmchecker_submissions',
+            array('id' => $previousAttempt->id)
+        );
+
+    }
+
+    public static function submit(\core\event\base $event) {
+        global $DB, $USER, $CFG;
+
+        $data = $event->get_data();
+        $submission = $DB->get_record($data['objecttable'],
+            array(
+                'id' => $data['objectid']
+        ));
+
+        $vmchecker_options = $DB->get_record('block_vmchecker_options',
+            array(
+                'assignid' => $submission->assignment
+        ));
```

Comment:
This can also be compressed.
```suggestion
        $vmchecker_options = $DB->get_record('block_vmchecker_options',
            array('assignid' => $submission->assignment));
```